src/modules/Library/features/AppletsCatalog/AppletsCatalog.tsx

```diff
@@ -30,41 +29,38 @@ export const AppletsCatalog = () => {
 
   useBreadcrumbs();
 
-  // const publishedApplets = library.usePublishedApplets();
-  const publishedApplets: PublishedAppletResponse = mockedPublishedAppletResponse;
+  const publishedApplets = library.usePublishedApplets();
 
   const [pageIndex, setPageIndex] = useState(0);
   const [recordsPerPage, setRecordsPerPage] = useState(DEFAULT_APPLETS_PER_PAGE);
-  const [searchText, setSearchText] = useState('');
+  const [search, setSearch] = useState('');
 
   const handleSearch = (searchText: string) => {
-    setSearchText(searchText);
+    setSearch(searchText);
     setPageIndex(0);
   };
 
   const handleChangePage = (event: unknown, newPage: number) => {
     setPageIndex(newPage);
-    // TODO: delete comment when endpoint is ready
-    // dispatch(library.thunk.getPublishedApplets({ recordsPerPage, pageIndex: newPage, searchText }));
+    dispatch(library.thunk.getPublishedApplets({ pageIndex: newPage, search }));
   };
 
   const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
     setRecordsPerPage(parseInt(event.target.value, 10));
     setPageIndex(0);
   };
 
-  const renderEmptyState = () => !!searchText && <EmptyState>{t('notFound')}</EmptyState>;
+  const renderEmptyState = () => !!search && <EmptyState>{t('notFound')}</EmptyState>;
 
   useEffect(() => {
-    // TODO: delete comment when endpoint is ready
-    // const timeout = setTimeout(
-    //   () =>
-    //     dispatch(library.thunk.getPublishedApplets({ recordsPerPage, pageIndex: 0, searchText })),
-    //   1000,
-    //   // TODO: discuss search - use hook or fix search by enter
-    // );
-    // return () => clearTimeout(timeout);
-  }, [searchText]);
+    const timeout = setTimeout(
```

Comment:
why do we need timeout?

Reply:
maybe need to use Debouncing?

Reply:
i will remove timeout. and debouncing has already added in search comp